API/Data/Repositories/UserRepository.cs

```diff
@@ -92,9 +92,11 @@ public interface IUserRepository
     Task<IList<DashboardStreamDto>> GetDashboardStreams(int userId, bool visibleOnly = false);
     Task<IList<AppUserDashboardStream>> GetAllDashboardStreams();
     Task<AppUserDashboardStream?> GetDashboardStream(int streamId);
+    Task<AppUserDashboardStream?> GetDashboardStreamWithUser(int streamId);
```

Comment:
You should use a SideNavIncludes pattern instead to allow one method to handle the extra join seemlessly.